lib/maropost/api.rb

```diff
@@ -1,93 +1,86 @@
 module Maropost
   class Api
-    def self.find(email)
-      response = request(:get,
-                         maropost_url('contacts/email.json', "contact[email]=#{CGI::escape(email)}"))
-      Maropost::Contact.new(JSON.parse response.body)
-    rescue RestClient::ResourceNotFound
-      nil
-    end
+    class << self
+      def find(email)
+        raise ArgumentError, 'must provide email' if email.blank?
+        response = request(
+          :get,
+          maropost_url('contacts/email.json', "contact[email]=#{CGI.escape(email)}")
+        )
+        Maropost::Contact.new(JSON.parse(response.body))
+      rescue RestClient::ResourceNotFound
+        nil
+      end
 
-    def self.update_subscriptions(contact)
-      if existing_contact = find(contact.email)
-        contact.id = existing_contact.id
-        update(contact)
-      else
-        create(contact)
+      def update_subscriptions(contact)
+        if existing_contact = find(contact.email)
```

Comment:
is it worth doing the same kind of presence check on these methods as you have on the find method?